proxy/txnmanager/transaction.go

```diff
@@ -0,0 +1,329 @@
+	if vars == nil || vars["token"] == "" {
+		http.Error(w, fmt.Sprintf("token not found: %v", vars), http.StatusInternalServerError)
+		return
+	}
+
+	token := vars["token"]
+
+	// decode the token
+	marshall, err := b64.URLEncoding.DecodeString(token)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to decode token: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	// unmarshall the token to get the json with all the informations
+	var content TransactionInfo
+	err = json.Unmarshal(marshall, &content)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to unmarshall token: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	// check if the transaction status is unknown
+	// if it is not unknown, it means that the transaction was already checked
+	if content.Status != UnknownTransactionStatus {
+		http.Error(w, "the transaction status is known", http.StatusBadRequest)
+		return
+	}
+
+	// get the signature as a crypto.Signature
+	signature, err := h.signer.GetSignatureFactory().SignatureOf(h.context, content.Signature)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to get Signature: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	// check if the hash is valid
+	if !h.checkHash(content.Status, content.TransactionID, content.LastBlockIdx, content.Time, content.Hash) {
+		http.Error(w, "invalid hash", http.StatusInternalServerError)
+		return
+	}
+
+	// check if the signature is valid
+	if !h.checkSignature(content.Hash, signature) {
+		http.Error(w, "invalid signaturee", http.StatusInternalServerError)
+		return
+	}
+
+	// check if if was submited not to long ago
+	if time.Now().Unix()-content.Time > int64(maxTimeTransactionCheck) {
+		// if it was submited to long ago, we reject the transaction
+		err = h.SendTransactionInfo(w, content.TransactionID, 0, RejectedTransaction)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("failed to send transaction info: %v", err), http.StatusInternalServerError)
+			return
+		}
+		return
+	}
+
+	// check if the transaction time stamp is possible
+	if time.Now().Unix()-content.Time < 0 {
+		http.Error(w, "the transaction is from the future", http.StatusInternalServerError)
+		return
+	}
+
+	// check if the transaction is included in the blockchain
+	newStatus, idx := h.checkTxnIncluded(content.TransactionID, content.LastBlockIdx)
+
+	// send the transaction info
+	err = h.SendTransactionInfo(w, content.TransactionID, idx, newStatus)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to send transaction info: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+}
+
+// checkHash checks if the hash is valid
+func (h *manager) checkHash(status TransactionStatus, transactionID []byte, LastBlockIdx uint64, Time int64, Hash []byte) bool {
+	// create the hash
+	hash := sha256.New()
+	hash.Write([]byte{byte(status)})
+	hash.Write(transactionID)
+	hash.Write([]byte(strconv.FormatUint(LastBlockIdx, 10)))
+	hash.Write([]byte(strconv.FormatInt(Time, 10)))
```

Comment:
Hashing should be a Fingerprint function that you call everywhere you want to compute the hash. That way you're sure it's always the hash of the same information